subsys/bluetooth/mesh/delayable_msg.c

```diff
@@ -0,0 +1,314 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/kernel.h>
+#include <errno.h>
+#include <stdlib.h>
+#include <zephyr/sys/slist.h>
+
+#include <zephyr/net/buf.h>
+#include <zephyr/bluetooth/mesh.h>
+
+#include "msg.h"
+#include "access.h"
+#include "net.h"
+
+#define LOG_LEVEL CONFIG_BT_MESH_ACCESS_LOG_LEVEL
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(bt_mesh_delayable_msg);
+
+static void delayable_msg_handler(struct k_work *w);
+static bool push_msg_from_delayable_msgs(void);
+
+static struct delayable_msg_chunk {
+	sys_snode_t node;
+	uint8_t data[CONFIG_BT_MESH_ACCESS_DELAYABLE_MSG_CHUNK_SIZE];
+} delayable_msg_chunks[CONFIG_BT_MESH_ACCESS_DELAYABLE_MSG_CHUNK_COUNT];
+
+static struct delayable_msg_ctx {
+	sys_snode_t node;
+	sys_slist_t chunks;
+	struct bt_mesh_msg_ctx ctx;
```

Comment:
Do we really need all the variables in this structure? It seems more appropriate to isolate the necessary variables individually.

Reply:
Transport layer mandates this structure.

Reply:
Indeed need this structure, but not all members.

Reply:
Knowledge of which variables are required and which aren't below the access layer is a violation of layer isolation. Then changing something for instance in the transport layer will cause change here also and etc. I guess this is an architectural mistake in the advantage of memory consumption.